### PR TITLE
Added `sort=` argument to all functions

### LINKED ISSUE #2: `tbl_adverse_events()` allow users to change default sorting
The SOC and AEs are sorted alphabetically. Should allow users to sort by frequency?

### PR DESCRIPTION
**Pull request by ddsjoberg**
@shannonpileggi can you take a look at this? what do you think of the user experience here? can be better?

- [ ] add unit tests
closes #2

### CODE REVIEW

File: R/tbl_ae.R [hunk 1]
```diff
@@ -33,6 +33,9 @@
 #' @param zero_symbol String used to represent cells with zero counts. Default
 #' is the em-dash (`"\U2014"`). Using `zero_symbol = NULL` will print the
 #' zero count statistics, e.g. `"0 (0)"`
+#' @param sort Controls order AEs and SOCs appear in the tables. Default is
+#'  `"alphanumeric"`. `sort = "frequency"` will sort in decreasing frequency
+#'  order.
```
Comment (shannonpileggi) [inline R/tbl_ae.R:38]: The `sort` argument applies only to ae overall (when ae is specified) or ae within soc (when both ae and soc are specified).  Maybe the documentation could be amended like:

```suggestion
#' @param sort Controls order AEs appear.  When `ae` is specified without `soc`, applies to AEs overall. When `ae` is #' specified with `soc`, applies to AEs within SOCs.  Default is
#'  `"alphanumeric"` for alphabetical sorting. `sort = "frequency"` will sort in decreasing frequency
#'  order.
```

**Review by shannonpileggi — CHANGES_REQUESTED** [review]
We should also add some tests, but this might be best after #67 is complete.

File: R/tbl_ae.R (hunk 1 above)
Comment (shannonpileggi) [reply R/tbl_ae.R:38]: Or I am wondering if I tested this enough to understand what was happening...

**Comment by shannonpileggi** [comment]
Other things I am thinking about, that might be worth a quick chat, is:

- would it be worthwhile to have sort_ae & sort_soc separate controls? or overkill?
- what happens if ae or soc's are input as a factor?  preserving that might be a nice feature if users have a specific order in mind

**Comment by ddsjoberg** [comment]
> * would it be worthwhile to have sort_ae & sort_soc separate controls? or overkill?
> * what happens if ae or soc's are input as a factor?  preserving that might be a nice feature if users have a specific order in mind

I think overkill to add two arguments for sorting. If the users have a specific ordering in mind, they can make the AE or SOC a factor, and the factor ordering is preserved.

**Comment by shannonpileggi** [comment]
Sounds good!  I was just checking a few combinations with this test set.

```
df2 <-
  tibble::tibble(
    patient_id = paste0("ID", 1:9),
    system_organ_class = c("Fruit", "Fruit", "Fruit", "Veg", "Veg", "Veg", "Veg", "Bread", "Bread"),
    adverse_event = c("apple",  "banana", "banana", "artichoke", "spinach", "spinach", "spinach", "ciabatta", "sourdough"),
    grade = rep(1, 9),
    trt = rep("A", 9)
  )
```

with no `sort` specified, factor levels are preserved.
```
  df2 %>%
    mutate(
      adverse_event = forcats::fct_relevel(adverse_event, "sourdough", "banana")
    ) %>%
    tbl_ae(
      id = patient_id,
      ae = adverse_event,
      statistic = "{n}"
    )
```

image

when `sort = "frequency"` is applied, sort overrides factor levels.

```
  df2 %>%
    mutate(
      adverse_event = forcats::fct_relevel(adverse_event, "sourdough", "banana")
    ) %>%
    tbl_ae(
      id = patient_id,
      ae = adverse_event,
      statistic = "{n}",
      sort = "frequency"
    )
```
image

when `sort = "alphanumeric"` the factor levels are preserved and alphanumeric does not override.

````
  df2 %>%
    mutate(
      adverse_event = forcats::fct_relevel(adverse_event, "sourdough", "banana")
    ) %>%
    tbl_ae(
      id = patient_id,
      ae = adverse_event,
      statistic = "{n}",
      sort = "alphanumeric"
    )
```

![image](https://user-images.githubusercontent.com/17747575/148831459-7aa975b8-d8f6-4251-a0d5-a6076c47ba24.png)

what should be the expected behavior here? to override, warn, or preserve factors?

**Comment by ddsjoberg** [comment]
alphanumeric is the default, and preserves the inherent ordering (be that a character vector or a factor). I think this is expected. Do you think it should be something else?

**Comment by shannonpileggi** [comment]
as a user, i might expect alphanumeric to overwrite factor ordering if it is present, so lets just add the behavior to the documentation!

**Comment by ddsjoberg** [comment]
> as a user, i might expect alphanumeric to overwrite factor ordering if it is present, so lets just add the behavior to the documentation!

ok sounds good. @shannonpileggi can you update it ?

**Comment by shannonpileggi** [comment]
not yet!  one  more thing, another option might be to call it `sort = "internal"` instead of alphanumeric to indicate it uses whatever internal mechanism it comes with (alphanumeric for character, factor levels for factor).

**Comment by ddsjoberg** [comment]
> not yet! one more thing, another option might be to call it `sort = "internal"` instead of alphanumeric to indicate it uses whatever internal mechanism it comes with (alphanumeric for character, factor levels for factor).

Hmmm, for me that is more confusing having a 3rd option. But I'll leave it up to you to add whatever your preference is! For me, factors are sorted according to their assigned order so I would expect that to persist in the function.

**Review by shannonpileggi — APPROVED** [review]
Ok, I updated docs a bit and added a few more lazy tests.  Let me know if you want any further revisions!

**Comment by shannonpileggi** [comment]
I think we need to test sorting a bit more on external data sets.

**Pull request merged**